src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts

```diff
@@ -0,0 +1,260 @@
+        if (formValues['name'] !== '' && formValues['type'] !== '') {
+          this.servicesHealthService.getInstalledPlugins(formValues['type']).subscribe(
+            (data: any) => {
+              this.plugins = data.plugins;
+              this.taskForm.get('plugin').setValue(this.plugins[0].name);
+            },
+            (error) => {
+              if (error.status === 0) {
+                console.log('service down ', error);
+              } else {
+                this.alertService.error(error.statusText);
+              }
+            });
+          this.getScheduleType();
+        }
+        break;
+      case 2:
+        nxtButton.textContent = 'Enable & Start Task';
+        if (formValues['name'] !== '' && formValues['plugin'].length > 0) {
+          this.isTaskAdded = true;
+          this.addScheduledTask(formValues, nxtButton);
+        }
+        break;
+      case 3:
+        nxtButton.textContent = 'Done';
+        if (this.taskId.length > 0) {
+          /** request started */
+          this.ngProgress.start();
+          this.schedulesService.enableSchedule(this.taskId).
+            subscribe(
+              () => {
+                /** request completed */
+                this.ngProgress.done();
+                this.isTaskEnabled = true;
+                this.enableTaskMsg = 'Task scheduled and enabled successfully.';
+                this.alertService.success(this.enableTaskMsg);
+                previousButton.disabled = true;
+              },
+              error => {
+                previousButton.disabled = false;
+                this.isTaskEnabled = false;
+                /** request completed */
+                this.ngProgress.done();
+                if (error.status === 0) {
+                  console.log('service down ', error);
+                } else {
+                  this.enableTaskMsg = error.statusText;
+                  this.alertService.error(error.statusText);
+                }
+              });
+        }
+        break;
+      case 4:
+        this.router.navigate(['/scheduled-task']);
+        break;
+      default:
+        break;
+    }
+
+    first.classList.remove('is-active');
+    first.classList.add('is-completed');
+
+    const sId = +id + 1;
+    const next = <HTMLElement>document.getElementById('' + sId);
+    if (next != null) {
+      next.setAttribute('class', 'step-item is-active');
+    }
+
+    const stepContent = <HTMLElement>document.getElementById('c-' + id);
+    if (stepContent != null) {
+      stepContent.classList.remove('is-active');
+    }
+
+    const nextContent = <HTMLElement>document.getElementById('c-' + sId);
+    if (nextContent != null) {
+      nextContent.setAttribute('class', 'box step-content is-active');
+    }
+  }
+
+  addScheduledTask(formValues, nxtButton) {
+    /** request started */
+    this.ngProgress.start();
+    formValues['schedule_enabled'] = false,
+    formValues['schedule_repeat'] = 30,
```

Comment:
why?

Reply:
then what should be the value of `schedule_repeat`? (as it is must for `INTERVAL` type schedule)

Reply:
ask on UI, keep default and placeholder value 30